lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -144,22 +144,22 @@ export class BrowserStorage extends CacheManager {
         switch (type) {
             case CacheSchemaType.ACCOUNT: {
                 const account = new AccountEntity();
-                return (CacheHelper.toObject(account, JSON.parse(value)) as AccountEntity);
+                return (CacheManager.toObject(account, JSON.parse(value)) as AccountEntity);
```

Comment:
I am getting an error on this `JSON.parse` call when there's an entry in the cache that is not in JSON format, e.g. a non-msal cache value. We probably need additional logic somewhere to ignore cache entries we don't expect

Reply:
I've added a fix for this code as discussed. The account filter code will now pass over any non-MSAL cache items.